yagt.md

```diff
@@ -0,0 +1,91 @@
+
+`draft` `optional`
+
+This NIP extends an existing indexable tag `g` and a new indexable tag `G`. 
+
+## Motivation
+The core aim of this NIP is to establish a unified and efficient method for geotagging within events. During the development of the [nostr-geotags](https://github.com/sandwichfarm/nostr-geotags) library which utilizes NIP-32, it became evident that more specific guidelines were needed for geotagging. This NIP is designed to provide a clear and standardized format for geotagging, promoting uniformity and effectiveness of geodata across nostr.
+
+## Rationale
+Developing a dedicated geotagging NIP offers several advantages over the general application of NIP-32 for geographical data:
+
+- **Simplicity and Clarity:** Using NIP-32 for geodata can lead to verbose and complex labels, making them challenging to understand. This NIP simplifies expression.
+- **Handling ISO Standards:** The complexity added by ISO-3166-3 country code changes when using NIP-32 is streamlined in this NIP by simply labeling the change as countryCode, thus making both publishing and filtering more straightforward.
+- **Significance of Geodata:** Given its ubiquity, importance and diversity, geodata warrants its own specific letter designations and detailed specifications within nostr.
+- **Extends Existing GeoTag** `NIP-52` defined the `g` geotag, which this NIP extends and is compatible with.
+
+## Schema 
+This NIP uses a similar pattern as `NIP-32` (`G` and `g` instead of `L` and `l`) but with an additional key as position `3` and catered geographical data.
+
+```
+["G", label] //optional
+["g", value, label, format]
+```
+
+## Keys
+Standards are assigned to human readable labels instead of labeling content with standards, which complicates both publishing and filtering. 
+
+- `geohash` - A geohash of any precision.
+- `countryCode` - `ISO-3166-1/3` country code. 
+- `countryName` - `ISO-3166-1/3` compatible country name. 
+- `regionCode` - `ISO-3166-2` region code. 
+- `regionName` - `ISO-3166-2` compatible region name.
+- `cityName` - The the internationally recognized name of the city, formally cased. Use multiple `cityName` tags if a city has multiple spellings. 
+- `cityCode` - _Placeholder, there is no city code standard_ 
+
+## Defaults
+When `g` tag has no `tag[2]` it is assumed to be of type (key) `geohash` 
+
+```json
+{ tags:
+  [
+    [ "g", "u2mwdd8q4" ]
+  ]
+}
+```
+
+## Examples
+
+### Minimal Example 
+```json
+{ tags:
+  [
+    [ "g", "u2mwdd8q4" ]
+  ]
+}
+```
+
+### Robust Example
+```json
+{ tags:
+  [
+    [ "G", "geohash" ],
+    [ "g", "u2mwdd8q4", "geohash"  ],
+    [ "g", "u2mwdd8q", "geohash"  ],
+    [ "g", "u2mwdd8", "geohash"  ],
+    [ "g", "u2mwdd", "geohash"  ],
+    [ "g", "u2mwd", "geohash"  ],
+    [ "g", "u2mw", "geohash"  ],
+    [ "g", "u2m", "geohash"  ],
+    [ "g", "u2", "geohash"  ],
+    [ "g", "u", "geohash"  ],
+    [ "G", "countryCode" ],
+    [ "g", "AI", "countryCode", "alpha-2" ],
```

Comment:
can we move the component descriptor to be last element on the array `[ "g", "AI", "alpha-2", "countryCode"]` ?

Reply:
Why?

Reply:
bc it is easy to parse the values ​​knowing that the last element is the component descriptor

Reply:
Other than making the specification both unwriteable and unreadable, this can already be achieved because the _label_ (component descriptor) is ALWAYS at `tag[2]`, as you have suggested it would only sometimes be at `tag[2]`. If you are requesting this because you have built a parser that makes this assumption, be aware that it will not work for the majority of tags that exist on nostr. We primarily use key positions.

But I digress, I was going to suggest removing the _format_ key altogether anyways, because there will be no collisions and the format can be assumed based on the string anyways. + removes bloat.

Keep the prior comments in mind though.

Reply:
Is now moot with recent update.